### PR TITLE
[967, 969, 970, 971]: Add ui for buying lands in listing modal

### LINKED ISSUE #967: Buy Land button within the Listing Process
Story: As a PO, I’d like users to have the option to buy land directly from LandWorks in order to increase number of listed properties.

We need to add the option for users to browse land from the listing process whenever the user doesn’t own land. The option should be available **only for Decentraland**. 
-	Add the sentence “Do you want to buy land and list it for rent?” in Step 1 of the listing process [as per design](https://www.figma.com/file/2b3IsJsAD3Pi4S9wwKD1zZ/Stilyan-LandWorks?node-id=771%3A786&t=OHUjV1HeZRKulZsu-4).
-	Add a “Buy Land” below the text. 

If user switches to Voxels and doesn't have land - no change in the listing modal.

AC: Given I don’t own any land and I open the listing process, then I should have a button to buy land directly from LandWorks regardless of the Metaverse I have selected.

### LINKED ISSUE #969: Browse land for sale within the listing process
Story: As a PO, I’d like users to be able to browse land for sale directly from LandWorks in order to increase number of listed properties.

1.	Once the user clicks the BUY LAND button from #XXX then land NFTs for sale should be displayed. The data for the listed NFTs should come from OpenSea because there is little to no volume for the collections we are interested in on other marketplaces.
2.	The land NFTs for sale should look the same way as the card components we have in the listing process so we should reuse the component we have in the listing process. The differences in the component are:
- In the last row of the component, we should have the price the cryptocurrency it’s listed and in $ next to it [as per design](https://www.figma.com/file/2b3IsJsAD3Pi4S9wwKD1zZ/Stilyan-LandWorks?node-id=771%3A1264&t=OHUjV1HeZRKulZsu-4).  There might be various currencies but that should be resolved by the API call. 
- If both prices (in crypto and in $) do not fit in the card component, hide the $ value as the crypto value is mor important. 
- In the right corner of the card component there should be an OpenSea icon to indicate that the listing is from OpenSea. Make sure the icon is clickable with a tooltip that says “View on OpenSea”. If the user clicks it – open the item in a new tab. If the user clicks anywhere else in the card it should indicate a selection of the card.
- Load 32 NFTs on load (4 full scrolls) in order to limit the calls. Load 32 more once the user reaches to the bottom.
3.	When the user is browsing NFTs we should have an extra component in Step 1 alongside the Metaverse selector and the type dropdown. The additional component should be a sort. 
- By default, the selection should be “Price: lowest first”.
- The only other available sorting options should be Price: highest first

### LINKED ISSUE #970: Summary in Step 5 of listing when buying land
In the last summary step - the only difference is that we need to have a "Buy and List Property" button rather than "Confirm" (as in the standard listing form).

Design: [here](https://www.figma.com/file/2b3IsJsAD3Pi4S9wwKD1zZ/Stilyan-LandWorks?node-id=1083%3A1115&t=OUg2Chj419mlXsov-4)

If user doesn't have enough funds in their wallet to complete the tx, they should have the buttons disabled and an [error message](https://www.figma.com/file/2b3IsJsAD3Pi4S9wwKD1zZ/Stilyan-LandWorks?node-id=1084%3A1678&t=OUg2Chj419mlXsov-4).

### LINKED ISSUE #971: Popup message in Step 5 when buying land
Add a confirmation popup message after user clicks the Buy and Rent button.
Design- [here](https://www.figma.com/file/2b3IsJsAD3Pi4S9wwKD1zZ/Stilyan-LandWorks?node-id=771%3A2876&t=OUg2Chj419mlXsov-4).

Decentraland in the text should be bolded
And the {price} placeholder to be changed with the actual price of the land the user is buying.

### PR DESCRIPTION
**Pull request by n0vax**
**Detailed description**:

**Which issue(s) this PR fixes**:
Fixes #967
Fixes #969
Fixes #970
Fixes #971

**Special notes for your reviewer**:

**Checklist**
- [ ] Documentation added
- [ ] Tests updated

### CODE REVIEW

File: src/modules/land-works/components/lands-explore-map/LandTooltip.tsx [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+  x: number;
+  y: number;
+  open?: boolean;
+}
+
+const LandTooltip = ({ land, open, x, y, container }: LandTooltipProps) => {
+  const history = useHistory();
+
+  const handlePopperClick = () => {
+    history.push({
```
[the remark this answers is not in the document]
Comment (failfmi) [reply src/modules/land-works/components/lands-explore-map/LandTooltip.tsx:38]: I think those are renamed to other files in dev branch and can be deleted.

File: src/hooks/useGetAssetsForBuyingQuery.tsx [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+
+  return {
+    assets,
+    continuation: data.continuation,
+  };
+};
+
+const url = `${config.reservoir.apiUrl}/orders/asks/v4?${contracts
+  .map((contract) => `contracts=${contract}`)
+  .join('&')}&source=opensea.io&sortBy=price`;
```
[the remark this answers is not in the document]
Comment (n0vax) [reply src/hooks/useGetAssetsForBuyingQuery.tsx]: What do you mean? Should I remove it?

File: src/modules/land-works/components/lands-explore-map/LandTooltip.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (n0vax) [reply src/modules/land-works/components/lands-explore-map/LandTooltip.tsx:38]: Its probably appeared after resolving merge conflicts

File: src/hooks/useGetCoinGeckoCoinBySymbol.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+const useGetCoinGeckoCoinBySymbol = (symbol: string) => {
+  const {
+    data: supportedCoins,
+    isLoading: areSupportedCoinsLoading,
+    error: errorSupportedCoins,
+  } = useGetCoinGeckoSupportedCoinsGroupedBySymbolQuery();
+  const supportedCoinsBySymbol = supportedCoins[symbol.toLowerCase()];
+
+  console.log(supportedCoinsBySymbol);
```
Comment (failfmi) [inline src/hooks/useGetCoinGeckoCoinBySymbol.ts:29]: nit: unnecessary log

**Review by failfmi — DISMISSED** [review]

**Review by failfmi — APPROVED** [review]

**Pull request merged**